### PR TITLE
feat: Support IQueryable projection mappings

### PR DESCRIPTION
**Pull request by latonz**
Closes https://github.com/riok/mapperly/issues/214

### CODE REVIEW

File: docs/docs/02-configuration/11-queryable-projections.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    // highlight-start
+    .ProjectToDto()
+    // highlight-end
+    .ToListAsync();
+```
+
+</TabItem>
+</Tabs>
+
+This is useful in combination with EntityFramework and other ORM solutions
```
Comment (CommonGuy) [inline docs/docs/02-configuration/11-queryable-projections.mdx:37]: Would change this to
> This is useful in combination with Entity Framework and other ORM solutions
which expose `IQueryable<T>`. Only fields present in the target class will be retrieved from the database.

The last sentence (code inlining) is not interesting to Mapperly consumers IMO, which is why I would remove it

File: docs/docs/02-configuration/11-queryable-projections.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+```csharp
+[Mapper]
+public static partial class CarMapper
+{
+    // highlight-start
+    public static partial IQueryable<CarDto> ProjectToDto(this IQueryable<Car> q);
+    // highlight-end
+
+    // highlight-start
+    [MapProperty(nameof(Car.Manufacturer), nameof(CarDto.Producer)]
```
Comment (CommonGuy) [inline docs/docs/02-configuration/11-queryable-projections.mdx:69]: Wouldn't it make more sense to support this on the projection method?

File: docs/docs/02-configuration/12-conversions.md [hunk 3]
```diff
@@ -47,3 +48,18 @@ public partial class CarMapper
   ...
 }
 ```
+
+## Enable only specific automatic conversions
+
+To enable only specific conversion types, set `EnabledConversions` the conversion type to enable:
+
+```csharp
+// this disables conversions using the ToString() method:
```
Comment (CommonGuy) [inline docs/docs/02-configuration/12-conversions.md:57]: Would change this to

> // This disables conversions using the ToString() method, which is enabled by default:

File: src/Riok.Mapperly/Diagnostics/DiagnosticDescriptors.cs [hunk 4]
```diff
@@ -229,4 +229,36 @@ internal static class DiagnosticDescriptors
...
+        "RMG030",
+        "Reference loop detected while mapping to an init only property",
+        "Reference loop detected while mapping from {0}.{1} to the init only property {2}.{3}, consider ignoring this property",
+        DiagnosticCategories.Mapper,
+        DiagnosticSeverity.Error,
+        true);
+
+    public static readonly DiagnosticDescriptor ReferenceLoopInCtorMapping = new DiagnosticDescriptor(
+        "RMG031",
+        "Reference loop detected while mapping to a ctor parameter",
```
Comment (CommonGuy) [inline src/Riok.Mapperly/Diagnostics/DiagnosticDescriptors.cs:251]: Would always spell out "constructor" instead of using the abbreviation

File: src/Riok.Mapperly/Descriptors/Mappings/QueryableProjectionMapping.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        ITypeSymbol sourceType,
+        ITypeSymbol targetType,
+        ITypeMapping delegateMapping) : base(sourceType, targetType)
+    {
+        _delegateMapping = delegateMapping;
+    }
+
+    public override IEnumerable<StatementSyntax> BuildBody(TypeMappingBuildContext ctx)
+    {
+        // #nullable disable
```
Comment (CommonGuy) [inline src/Riok.Mapperly/Descriptors/Mappings/QueryableProjectionMapping.cs:31]: Why is this needed here? Add a comment. May also need to be added to the documentation

File: docs/docs/02-configuration/11-queryable-projections.mdx (hunk 1 above)
Comment (latonz) [reply docs/docs/02-configuration/11-queryable-projections.mdx]: Improved.

File: docs/docs/02-configuration/11-queryable-projections.mdx (hunk 2 above)
Comment (latonz) [reply docs/docs/02-configuration/11-queryable-projections.mdx:69]: https://www.itrockt.ch/mitglieder/riok-gmbh/

File: src/Riok.Mapperly/Descriptors/Mappings/QueryableProjectionMapping.cs (hunk 5 above)
Comment (CommonGuy) [reply src/Riok.Mapperly/Descriptors/Mappings/QueryableProjectionMapping.cs:32]: I don't think just disabling nullability solves this problem correctly. What happens if values are actually null?

Don't we need to handle all nullable values correctly anyway? I don't quite get why this is needed
